crates/protocol/hardforks/src/ecotone.rs

```diff
@@ -105,6 +106,22 @@ impl Ecotone {
 
     /// Returns the list of [TxDeposit]s for the Ecotone network upgrade.
     pub fn deposits() -> impl Iterator<Item = TxDeposit> {
+        // Verify source has of L1Block
+        // See: <https://specs.optimism.io/protocol/ecotone/derivation.html#l1block-deployment>
+        assert_eq!(
```

Comment:
can we just do this when running with debug profile and otherwise not add the perf overhead. unsure we would want this to panic in production regardless.
```suggestion
        debug_assert!(
```